appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -72,35 +78,43 @@ export async function waitForDeploymentToComplete(client: SiteClient, kuduClient
                 return;
             }
         } else {
-            await new Promise((resolve: () => void): void => { setTimeout(resolve, pollingInterval); });
+            await delay(pollingInterval);
         }
     }
 }
 
-async function getLatestDeployment(kuduClient: KuduClient, permanentId: string | undefined, initialStartTime: Date | undefined): Promise<[DeployResult | undefined, string | undefined, Date | undefined]> {
+async function tryGetLatestDeployment(kuduClient: KuduClient, permanentId: string | undefined, initialStartTime: Date | undefined, expectedId?: string): Promise<[DeployResult | undefined, string | undefined, Date | undefined]> {
     let deployment: DeployResult | undefined;
-    if (permanentId) {
-        // Use "permanentId" to find the deployment during its "permanent" phase
-        deployment = await kuduClient.deployment.getResult(permanentId);
-    } else if (initialStartTime) {
-        // Use "initialReceivedTime" to find the deployment during its "temp" phase
-        deployment = (await kuduClient.deployment.getDeployResults())
-            // tslint:disable-next-line:no-non-null-assertion
-            .filter((deployResult: DeployResult) => deployResult.startTime && deployResult.startTime >= initialStartTime!)
-            .sort((a: DeployResult, b: DeployResult) => nonNullProp(b, 'startTime').valueOf() - nonNullProp(a, 'startTime').valueOf())
-            .shift();
-        if (deployment && !deployment.isTemp) {
-            // Make note of the id once the deplyoment has shifted to the "permanent" phase, so that we can use that to find the deployment going forward
-            permanentId = deployment.id;
-        }
-    } else {
-        // Use "latest" to get the deployment before we know the "initialReceivedTime" or "permanentId"
-        deployment = <DeployResult | undefined>await kuduClient.deployment.getResult('latest');
-        if (deployment && deployment.startTime) {
-            // Make note of the startTime because that is when kudu has began the deployment process,
-            // so that we can use that to find the deployment going forward
-            initialStartTime = deployment.startTime;
+    try {
+        if (permanentId) {
+            // Use "permanentId" to find the deployment during its "permanent" phase
+            deployment = await kuduClient.deployment.getResult(permanentId);
+        } else if (expectedId) {
+            // if we have a "expectedId" we know which deployment we are looking for, so wait until latest id reflects that
+            const latestDeployment: DeployResult = await kuduClient.deployment.getResult('latest');
+            permanentId = latestDeployment.id === expectedId ? latestDeployment.id : undefined;
```

Comment:
I don't see you setting `deployment` in this case

Reply:
I didn't intentionally since it will be set the next time the command is called (due to having a `permanentId` now).  I guess this could cause a 5 second delay for output though so I'll change that.